pact4s/src/test/scala/org/broadinstitute/dede/workbench/leonardo/provider/LeoProvider.scala

```diff
@@ -0,0 +1,169 @@
+  implicit val loggerIO: StructuredLogger[IO] = Slf4jLogger.getLogger[IO]
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+  implicit val system: ActorSystem = ActorSystem("leotests")
+
+  val mockOpenIDConnectConfiguration: OpenIDConnectConfiguration = mock[OpenIDConnectConfiguration]
+  val mockStatusService: StatusService = mock[StatusService]
+  val mockProxyService: ProxyService = mock[ProxyService]
+  val mockRuntimeService: RuntimeService[IO] = mock[RuntimeService[IO]]
+  val mockDiskService: DiskService[IO] = mock[DiskService[IO]]
+  val mockDiskV2Service: DiskV2Service[IO] = mock[DiskV2Service[IO]]
+  val mockAppService: AppService[IO] = mock[AppService[IO]]
+  val mockRuntimeV2Service: RuntimeV2Service[IO] = mock[RuntimeV2Service[IO]]
+  val mockAdminService: AdminService[IO] = mock[AdminService[IO]]
+  val mockContentSecurityPolicyConfig: ContentSecurityPolicyConfig = mock[ContentSecurityPolicyConfig]
+  val refererConfig: RefererConfig = RefererConfig(Set("*"), enabled = true)
+  val mockUserInfoDirectives: MockUserInfoDirectives = new MockUserInfoDirectives {
+    override val userInfo: UserInfo = defaultUserInfo
+  }
+
+  val routes =
+    new HttpRoutes(
+      mockOpenIDConnectConfiguration,
+      mockStatusService,
+      mockProxyService,
+      mockRuntimeService,
+      mockDiskService,
+      mockDiskV2Service,
+      mockAppService,
+      mockRuntimeV2Service,
+      mockAdminService,
+      mockUserInfoDirectives,
+      mockContentSecurityPolicyConfig,
+      refererConfig
+    )
+
+  private val providerStatesHandler: StateManagementFunction = StateManagementFunction {
+    case ProviderState(States.AppExists, _) =>
+      when(mockAppService.getApp(any[UserInfo], any[CloudContext.Gcp], AppName(anyString()))(any[Ask[IO, AppContext]]))
+        .thenReturn(IO {
+          GetAppResponse(
+            None,
+            AppName("exampleApp"),
+            CloudContext.Gcp(GoogleProject("exampleProject")),
+            RegionName("exampleRegion"),
+            KubernetesRuntimeConfig(NumNodes(8), MachineTypeName("exampleMachine"), autoscalingEnabled = true),
+            List.empty[AppError],
+            AppStatus.Unspecified,
+            Map.empty[KubernetesSerializableName.ServiceName, URL],
+            Some(DiskName("exampleDiskName")),
+            Map.empty[String, String],
+            AuditInfo(WorkbenchEmail(""), Instant.now(), None, Instant.now()),
+            AppType.CromwellRunnerApp,
+            ChartName(""),
+            None,
+            Map.empty[String, String]
+          )
+        })
+    case _ =>
+      loggerIO.debug("other state")
+  }
+
+  val provider: ProviderInfoBuilder =
+    ProviderInfoBuilder(name = "leonardo",
+                        pactSource = PactSource
+                          .FileSource(
+                            Map("aou" -> new File("./src/test/resources/aou-leonardo.json"))
```

Comment:
Reference to the pact in the repo. This is not ideal to have this file reference in here, because it currently only runs correctly when you run the leo-consumer-contract-tests job and not when you run LeoProvider.scala directly.